package swarm.msgs;

/// \ingroup swarm_msgs
/// \interface Datagram
/// \brief A message containing some payload to communicate among one or
/// multiple teammates.

message Datagram
{
  /// \brief Address of the sender.
  required uint32 src_address = 1;

  /// \brief Address of the destination.
  required uint32 dst_address = 2;

  /// \brief Destination port.
  required uint32 dst_port    = 3;

  /// \brief Payload.
  required string data        = 4;
}
